scripts/gen_ldelf_hex.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: BSD-2-Clause
+#
+# Copyright (c) 2019, Linaro Limited
+#
+
+from __future__ import print_function
+from __future__ import division
+
+import argparse
+import sys
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import SymbolTableSection
+from elftools.elf.constants import P_FLAGS
+import struct
+import re
+from collections import deque
+
+
+def round_up(n, m):
+    if n == 0:
+        return 0
+    else:
+        return (((n - 1) // m) + 1) * m
+
+
+def emit_load_segments(elffile, outf, prefix):
+    load_size = 0
+    n = 0
+    for segment in elffile.iter_segments():
+        if segment['p_type'] == 'PT_LOAD':
+            if n == 0:
+                if segment['p_flags'] != (P_FLAGS.PF_R | P_FLAGS.PF_X):
+                    print('Expected first load section to be read/execute')
```

Comment:
s/section/segment/ (and same below, twice).

Reply:
I'll fix